### DISCUSSION TITLE
Generic 2nd serial port

### DISCUSSION THREAD

**Discussion by Minims** [Desuuuu/klipper#76] [Q&A] [answered]
Hello,

I was trying to use this Klipper fork to enable a second serial on my board.
The goal is to use this second serial port for something else, like an ESP8266, or a screen that already communicate on RX/TX with his own firmware to send Gcode.( It's the case for BTT TFT, MKS TFT)

Thanks.

**Comment by Desuuuu** [comment, marked as answer]
The [runtime-uart](https://github.com/Desuuuu/klipper/tree/runtime-uart) and [dgus-display](https://github.com/Desuuuu/klipper/tree/dgus-display) branches both have the modifications to use serial ports dynamically.

If you intend to use a serial port for anything other than a DGUS T5UID1 touchscreen, you'll need to write your own Klippy module to support your usecase.

You can have a look at the [bus module](https://github.com/Desuuuu/klipper/blob/b0c5e941a71eb655602f67cf093435895ed04d5d/klippy/extras/bus.py#L225) for reference and the [dgus_display module](https://github.com/Desuuuu/klipper/tree/dgus-display/klippy/extras/dgus_display) is an example use of it.
